contracts/cw721-metadata-onchain/src/lib.rs

```diff
@@ -4,11 +4,29 @@ use serde::{Deserialize, Serialize};
 use cosmwasm_std::Empty;
 pub use cw721_base::{ContractError, InstantiateMsg, MintMsg, MinterResponse, QueryMsg};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-pub struct Extension {
-    pub metadata_uri: String,
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Trait {
+    pub display_type: Option<String>,
+    pub trait_type: String,
+    pub value: String,
 }
 
+// see: https://docs.opensea.io/docs/metadata-standards
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Metadata {
+    pub image: Option<String>,
+    pub image_data: Option<String>,
+    pub external_url: Option<String>,
+    pub description: Option<String>,
+    pub name: Option<String>,
+    pub attributes: Option<Vec<Trait>>,
+    pub background_color: Option<String>,
+    pub animation_url: Option<String>,
+    pub youtube_url: Option<String>,
+}
+
+pub type Extension = Option<Metadata>;
```

Comment:
The only reason to use this over cw721-base is to use Metadata. I would make it required. (not an Option)

Reply:
main reason is to interoperate with cw721-base better, so people can use `NftInfoResponse<Extension>` from `cw721-metadata-onchain` to query both cw721-base and cw721-metadata-onchain